### ISSUE TITLE
[Feature]: Reverse proxy documentation and examples (NPM etc.)

### ISSUE THREAD

**Issue by ollioddi** [immich-app/immich#1528]
### Feature detail

Hey.

I would like to put Immich behind a reverse proxy (In my case Nginx Proxy Manager), but find it a little bit hard to figure out how to.

Does anyone have examples for configuration with docker compose?

I found this thread on [reddit](https://www.reddit.com/r/selfhosted/comments/wp1zxd/immich_and_nginx_proxy_manager/) with various tips.

It seems like you shouldn't disable the immich docker network, since it uses it internally but i would like to know the following:

- How can i set a static IP on a custom VLAN
- How should NPM handle the different /api endpoint

I am using Unraid with docker-compose and have Immich running and several Docker networks with different VLANS.

A suggestion would be to add examples to the docs with various reverse proxies.

Let me know if this doesn't fit as a feature request.

Thanks in advance!

### Platform

Server

**Comment by jrasm91** [comment]
Correct, we even have a section (placeholder) for it already: "Guides".

For npm, which is what I use, I just update the immich-proxy container to also use the docker network that the npm container uses. Then the IP/hostname you use for the upstream server is `immich-proxy:8080`

If all the immich containers are on the same network, then the only "externally available" ip/port you'd need would be proxy container + port 2283. If you hook up the proxy container to the npm network you don't even need that, as you can use the internal service name + port instead.

![image](https://user-images.githubusercontent.com/4334196/216461305-90dd7c39-3a28-46e3-9b6f-7d875f4a8394.png)
